api/envoy/extensions/transport_sockets/tls/v3/tls.proto

```diff
@@ -124,8 +125,24 @@ message DownstreamTlsContext {
   OcspStaplePolicy ocsp_staple_policy = 8 [(validate.rules).enum = {defined_only: true}];
 }
 
+// TLS key log configuration.
+// The key log file format is "format used by NSS for its SSLKEYLOGFILE debugging output" (text taken from openssl man page)
+message TlsKeyLog {
+  // The path to save the TLS key log.
+  string tls_keylog_path = 1 [(validate.rules).string = {min_len: 1}];
+
+  // The src IP address that will be used to filter the connection which should save the TLS key log
+  // If it is not set, any src IP address will be matched.
+  config.core.v3.Address tls_keylog_src = 2;
```

Comment:
See https://github.com/envoyproxy/envoy/pull/19182/files#r788966910

Reply:
@ggreenway , I do not think we should use CidrRange since this feature is only for troubleshooting. When we do troubleshooting, we always focus on one specify src or dst to make things simpler.

Reply:
> @ggreenway , I do not think we should use CidrRange since this feature is only for troubleshooting. When we do troubleshooting, we always focus on one specify src or dst to make things simpler.

For your specific use case, that's how you want to use it. But configuring as a CidrRange is a more general API that will cover more use cases. And nearly all the code to handle it is already written: there's code to take the CidrRange and make a Trie out of it, and code to see if a given IP matches or not.

Reply:
fixed